### ISSUE TITLE
No logging option?

### ISSUE THREAD

**Issue by morsmodre** [aeroxis/sultan#24]
Hi ~ First of all: this is awesome!
Second, is there a way to remove sultan logging?

txtx!

**Comment by davydany** [comment]
Glad to hear that you like Sultan! 

Sultan uses the standard Python logging library that comes with the standard library, so you can simply override the logger for Sultan like this:


```python
import logging
from sultan.api import Sultan

# configure your logger first
logger = logging.getLogger('sultan')
logger.setLevel(logging.ERROR)

# use sultan like you normally do
with Sultan.load() as s:
    s.ls('-lah /tmp').run()
```

Typically you'd just need to perform the logging configuration at wherever your application starts running. 

You can also use Dict Config to configure your logger like this: http://python-guide-pt-br.readthedocs.io/en/latest/writing/logging/#example-configuration-via-a-dictionary OR https://docs.djangoproject.com/en/1.11/topics/logging/#examples

And so, for Sultan, you'd do something like this: 

```python
import logging
from logging.config import dictConfig

# configure the logger first
config = {
    "version": 1,
    "loggers": {
        "sultan": { "level": "ERROR" }
    }
}

dictConfig(config)

# use sultan like you normally do
with Sultan.load() as s:
    s.ls('-lah /tmp').run()
```

**Comment by davydany** [comment]
If that resolves it, please let me know and I'll close the ticket.

**Comment by morsmodre** [comment]
Unfortunatly this doesn't work. With this:
`# Redefine logging
import logging
# configure your logger first
logger = logging.getLogger('sultan')
print logger.level

with Sultan.load() as s:
    s.ls('-lah /tmp').run()

logger.setLevel(logging.ERROR)
print logger.level

with Sultan.load() as s:
    s.ls('-lah /tmp').run()
`
I get this:
`
0
[sultan]: ls -lah /tmp;
40
[sultan]: ls -lah /tmp;
`
So he basically sets the log level, but still does it ~ any more ideas?

txtx!

**Comment by davydany** [comment]
Alright, looks like I need to make changes to the code. Unfortunately, I'm not near my laptop to code this right now, but I'll fix this and update the code on PyPi before this weekend, and have it ready. 

Thanks for bringing this issue up!

**Comment by davydany** [comment]
Took a while, and I apologize for that, but I added the ability to disable logging all together. 

First, upgrade to `sultan==0.5`.

```bash
pip install --upgrade sultan
```

And disable logging like this:

```python
with Sultan.load(logging=False) as s:
    s.ls('/tmp').run() # or whatever set of commands you want to run
```

**Comment by davydany** [comment]
@morsmodre Let me know if this works.

**Comment by davydany** [comment]
Documentation on ReadTheDocs: http://sultan.readthedocs.io/en/latest/sultan-examples.html#example-9-disable-logging

**Comment by davydany** [comment]
Since this has been implemented, and available in version `0.5+`, I'm going to close this ticket.
